README.md

```diff
@@ -1,26 +1,55 @@
-# Offscreen Effect Player module
+# Offscreen Effect Player module (OEP-module)
+
+[Description of the **Offscreen Effect Player** on the Banuba SDK website](https://docs.banuba.com/face-ar-sdk-v1/core/cpp_api_overview#offscreen-effect-player)
 
 ## Description
 
-Common submodule for some projects using **offscreen_effect_player**
-The module implements interaction with the effect player. The module consists of six main interfaces:
+Common submodule for projects integrating Banuba SDK using C++. This module is independent of the Banuba SDK and represents the implementation of interfaces for  [**Offscreen effect player**](./interfaces/offscreen_effect_player.hpp). The module is a wrapper over Banuba SDK interfaces and provides necessary interfaces for image processing including postprocessing (orientation change, format conversion), and management of effects. This module demonstrates one of the approaches and simplifies integration to the application.
+
+## Folders structure
+
+- [**docs**](./docs/) - illustrations with screenshots and images
+- [**interfaces**](./interfaces/) - contains the declaration of offscreen effect player
+- [**offscreen_effect_player**](./offscreen_effect_player/) - contains the implementation of the **offscreen_effect_player**, **image_processing_result** and **pixel_buffer** intefaces. The implementation of **offscreen_effect_player** manages the rendering via the **ofscreen_render_target** interface and manages **effect_player** providing the main API for image processing by the Banuba SDK.
+- [**offscreen_render_target**](./offscreen_render_target/) - contains the implementation for the **offscreen_render_target** interface. The purpose of this submodule is to provide and manage the graphical context for offscreen rendering. By default, implements OpenGL but can be overloaded on the application level to use other rendering engines. The current implementation is responsible for preparing OpenGL framebuffers and textures for rendering and frame postprocessing (the resulted image conversions and transformations).
+- [**opengl**](./opengl/) - OpenGL utilities used by **offscreen_render_target** interface implementation
+- [**third**](./third/) - third party libraries
+
+## Description of interfaces
+
+The module by itself implements a piece of image processing conveyor: it accepts image for processing, confgures the offscreen rendering context, pass the image further to the main rendering engine - the Banuba SDK, and after provides the capabilities for image postprocessing. Based on this the module implements only some of the interfaces which do not introduce specifics of a particular API, e.g. Banuba SDK supports [**C++ API**](https://docs.banuba.com/face-ar-sdk-v1/core/cpp_api_overview) and [**C API**](), the module doesn't know which API is used for the Banuba SDK management that is why effect_player interface is left unimplemented and its implementation is provided by the application.
```

Comment:
This module is a part of the image processing flow: it accepts images for processing, configures the offscreen rendering context, passes the images further to the main rendering engine of the Banuba SDK, and provides the capabilities for image postprocessing afterwards. Based on this, the module only implements some of the interfaces which don't include the specifics of a particular API. For example, Banuba SDK supports [**C++ API**](https://docs.banuba.com/face-ar-sdk-v1/core/cpp_api_overview) and [**C API**](), but the module isn't dependent on any of them. That's why effect_player interface is left unimplemented, and...